main.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	log "github.com/Sirupsen/logrus"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/zlint"
+	"os"
+	"runtime"
+	"sync"
+	"github.com/zmap/zlint/lints"
+)
+
+var ( //flags
+	inPath           string
+	outPath          string
+	numCertThreads   int
+	prettyPrint      bool
+	numProcs         int
+	channelSize      int
+	crashIfParseFail bool
+	format           string
+)
+
+func init() {
+	flag.StringVar(&inPath, "input-file", "", "File path for the input certificate(s).")
+	flag.StringVar(&outPath, "output-file", "-", "File path for the output JSON.")
+	flag.BoolVar(&prettyPrint, "list-lints-json", false, "Use this flag to print supported lints in JSON format, one per line")
+	flag.IntVar(&numCertThreads, "cert-threads", 1, "Use this flag to specify the number of threads in -threads mode.  This has no effect otherwise.")
+	flag.IntVar(&numProcs, "procs", 0, "Use this flag to specify the number of processes to run on.")
+	flag.IntVar(&channelSize, "channel-size", 1000, "Use this flag to specify the number of values in the buffered channel.")
+	flag.BoolVar(&crashIfParseFail, "fatal-parse-errors", false, "Fatally crash if a certificate cannot be parsed. Log by default.")
+	flag.StringVar(&format, "format", "pem", "one of {pem, base64}")
+	flag.Parse()
+}
+
+func scannerSplitPEM(data []byte, atEOF bool) (int, []byte, error) {
+	block, rest := pem.Decode(data)
+	if block != nil {
+		size := len(data) - len(rest)
+		return size, data[:size], nil
+	}
+	return 0, nil, nil
+}
+
+func appendZLintToCertificate(cert *x509.Certificate, lintResult *lints.LintReport) ([]byte, error){
+	return json.Marshal(struct {
+		Raw	[]byte		   `json:"raw"`
+		Parsed  *x509.Certificate  `json:"parsed"`
+		ZLint	*lints.LintReport  `json:"zlint"`
+	}{
+		Raw:	cert.Raw,
+		Parsed:	cert,
+		ZLint:	lintResult,
+	})
+}
+
+func ProcessCertificate(in <-chan []byte, out chan<- []byte, wg *sync.WaitGroup) {
+	log.Info("Processing certificates...")
+	defer wg.Done()
+	for raw := range in {
+		parsed, err := x509.ParseCertificate(raw)
+		if err != nil { //could not parse
+			if crashIfParseFail {
+				log.Fatalf("could not parse certificate with error: %v", err)
+			} else {
+				log.Warnf("could not parse certificate with error: %v", err)
```

Comment:
Passing an `error` object as a `%s` results in `err.Error()` being printed, rather than a dump of the variables in the Go object that implements the `error` interface.